.github/workflows/macos_ci.yml

```diff
@@ -0,0 +1,69 @@
+name: ACCP macOS CI
+
+# Workflow syntax.
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+on:
+  pull_request:
+    branches:
+      - '*'
+
+permissions:
+  contents: read
+
+env:
+  PACKAGE_NAME: ACCP
+
+jobs:
+  # Job to run regular ACCP tests on MacOS.
+  macOS:
+    runs-on: macos-11 # latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    # Set up Corretto 8. The JDK version should be least 10 for a regular ACCP build,
+    # but JAVA_HOME will be overwritten with subsequent versions we set up, so we save
+    # the path here beforehand.
+    - name: Set up corretto 8
+      uses: actions/setup-java@v3
+      with:
+        java-version: '8'
+        distribution: 'corretto'
+    - name: Get JAVA_HOME variable for correto 8
+      run: |
+        echo "JAVA8_HOME=${{ env.JAVA_HOME }}" >> $GITHUB_ENV
+
+    # Test on Corretto 11.
+    - name: Set up corretto 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'corretto'
+    - name: Build and run tests for ${{ env.PACKAGE_NAME }} in corretto 11
+      env:
+        TEST_JAVA_HOME: ${{ env.JAVA_HOME }}
+      run: |
+        ./tests/ci/run_accp_basic_tests.sh
+        ./tests/ci/run_accp_test_integration.sh
+
+    # Test on Corretto 8.
+    - name: Build and run tests for ${{ env.PACKAGE_NAME }} in corretto 8
```

Comment:
nit: move this Corrtto 8 testing up closer to the setup?

also, this is a lack of GitHub CI familiarity on my part, but why do you need to break out the various setup and test tasks into separate jobs? also, is each `name`d job run sequentially or concurrently?

Reply:
Moved closer 👍 
`macOS` is actually the only job that's being run. Each job is points to a single Github Action Webhook. The `name`s actually refer to each `step` within the `job`, so they're run sequentially. I've looked into trying to run them concurrently, but `1 job=1 webhook`, so it bloats up the CI.